src/client/dfuse/dfuse.h

```diff
@@ -0,0 +1,746 @@
+/**
+ * (C) Copyright 2016-2019 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#ifndef __DFUSE_H__
+#define __DFUSE_H__
+
+#include <fuse3/fuse.h>
+#include <fuse3/fuse_lowlevel.h>
+
+#include <gurt/list.h>
+#include <gurt/hash.h>
+
+#include "dfuse_cnss.h"
+#include "dfuse_gah.h"
+#include "dfuse_fs.h"
+#include "dfuse_bulk.h"
+#include "dfuse_pool.h"
+
+/**
+ * A common structure for holding a cart context and thread details.
+ *
+ * This is included in both dfuse_state for global values, and once per
+ * projection for projection specific entries.
+ */
+struct dfuse_ctx {
+	/** cart context */
+	crt_context_t			crt_ctx;
```

Comment:
True, but we need to replace it with something else, so it needs to be here as a placeholder.  When it comes to inserting DAOS calls this will be the first thing to go.

Reply:
This struct is what is allocated per thread, to allow multi-threaded (cart) progress, so something that will need to be looked at but I don't want to replace this yet until we have the ability to run/test it.

Reply:
i just meant the crt_context_t struct, not the dfuse_ctx.

Reply:
I've done this in https://github.com/daos-stack/daos/pull/343 if you think that's the way to go I can squash that into this PR.